File: api/battle_sys/sp_counter.py
class SpCounter:
    def __init__(self):
        self.sp = 3

    def increase(self):
        if self.sp < 5:
            self.sp += 1
        elif self.sp == 5:
            self.sp == 5

    def decrease(self):
        if self.sp > 0:
            self.sp -= 1
        elif self.sp == 0:
            self.sp = 0


# test = SpCounter()
# test.increase()
# print(test.sp)
# test.decrease()
# print(test.sp)

File: ghi/src/battle_sys/sp_counter.mjs
class SpCounter{
    constructor(){
        this.sp = 3
    }
    increase(){
        if(this.sp < 5){
        this.sp += 1
        }
    }
    decrease(){
        if(this.sp > 0){
            this.sp -= 1
        }
    }
}


export default SpCounter
